std/os.zig

```diff
@@ -2166,6 +2170,28 @@ pub fn args() ArgIterator {
 
 /// Caller must call argsFree on result.
 pub fn argsAlloc(allocator: *mem.Allocator) ![]const []u8 {
+    if (builtin.os == Os.wasi) {
+        var count: usize = undefined;
+        var buf_size: usize = undefined;
+
+        _ = os.wasi.args_sizes_get(&count, &buf_size);
```

Comment:
This should assert that there is no error, or handle it. If it asserts that there is no error, it should explain why the error is impossible.

Reply:
The return values are not documented at all. I looked at the source code of wasmtime and Lucet  and they are not consistent. I have introduced error.ArgsSizesGetFailed and error.ArgsGetFailed in the meantime.

Reply:
In that case we can follow the pattern set by std/os.zig where an unexpected/undocumented error code is returned. Look for unexpectedErrorPosix and unexpectedErrorWindows.

Generally it's accepted that an undocumented error code might be returned from the OS which translates to error.Unexpected and has a mechanism for logging error codes with a build option.

Reply:
I've changed them to unexpectedErrorPosix. Most of WASI functions (if they are applicable), are similar to POSIX.